keyring.php

```diff
@@ -133,12 +133,19 @@ static function request_handlers() {
 		// Some services do not return the redirect_uri params, in this case these are packed into the state param.
 		// If so, they are unpacked again here and merged back into the request globals.
 		if ( ! empty( $_REQUEST['state'] ) ) {
-			$unpacked_state = unserialize( base64_decode( $_REQUEST['state'] ) );
-			if ( ! empty( $unpacked_state['action'] )) {
-				foreach ( $unpacked_state as $key => $value) {
+			$unpacked_state =  unserialize( base64_decode( $_REQUEST['state'] ) );
+			
+			if ( ! empty( $unpacked_state['hash'] ) ) {
+				if ( ! $validated_parameters = Keyring_Util::get_validated_parameters( $unpacked_state ) ) {
```

Comment:
Should we combine these checks? Something like:

```
if ( empty( $unpacked_state['hash'] ) ||
   ! $validated_parameters = Keyring_Util::get_validated_parameters( $unpacked_state ) ) {
```

There's the unlikely situation where `state` isn't empty but it doesn't unserialize properly and I don't think we'll show an error in that circumstance.

Reply:
I'll need to check what coding standards we're working to as well, but if it's the WordPress ones, I'm not sure we're allowed to do the assignment in the `if` statement.

Reply:
State is actually used by default with most of the services already, so it will likely always be set, but will only deserialize and include a 'hash' param for instagram at the moment, so if 'hash' is empty we want to bypass this flow.

Reply:
> I'm not sure we're allowed to do the assignment in the `if` statement.

turns out we're not - fixed.